service/src/main/java/bio/terra/profile/service/azure/ApplicationService.java

```diff
@@ -16,12 +18,53 @@ public ApplicationService(CrlService crlService) {
     this.crlService = crlService;
   }
 
+  /**
+   * Retrieves the managed applications for a given Azure subscription
+   *
+   * @param subscriptionId Azure subscription ID to be queried
+   * @return List of managed applications present in the subscription
+   */
   public Stream<Application> getApplicationsForSubscription(UUID subscriptionId) {
     var appMgr = crlService.getApplicationManager(subscriptionId);
     return appMgr.applications().list().stream();
   }
 
+  /**
+   * Retrieves the tenant associated with a given Azure subscription ID
+   *
+   * @param subscriptionId Azure subscription ID to be queried
+   * @return UUID for the associated tenant
+   */
   public UUID getTenantForSubscription(UUID subscriptionId) {
     return crlService.getTenantForSubscription(subscriptionId);
   }
+
+  /**
+   * Adds a tag to the MRG associated with a given profile. If the tag is already present, * throws
+   * a DuplicateTagException.
```

Comment:
typo
```suggestion
   * Adds a tag to the MRG associated with a given profile. If the tag is already present, throws
   * a DuplicateTagException.
```
Or, you could move that part of the javadoc to a `@throws` clause below.